parsl/executors/status_handling.py

```diff
@@ -52,25 +52,25 @@ def status_polling_interval(self):
             return self._provider.status_polling_interval
 
     @abstractmethod
-    def _get_job_ids(self) -> List[object]:
+    def _get_block_and_job_ids(self) -> Tuple[List[object], List[object]]:
         raise NotImplementedError("Classes inheriting from StatusHandlingExecutor must implement "
-                                  "_get_job_ids()")
+                                  "_get_block_and_job_ids()")
 
-    def _fail_job_async(self, job_id: Any, message: str):
+    def _fail_job_async(self, block_id: Any, message: str):
         """Marks a job that has failed to start but would not otherwise be included in status()
         as failed and report it in status()
         """
-        if job_id is None:
-            job_id = "failed-block-{}".format(self._generated_job_id_counter)
-            self._generated_job_id_counter += 1
-        self._simulated_status[job_id] = JobStatus(JobState.FAILED, message)
+        if block_id is None:
```

Comment:
Can block_id ever be None here?

Reply:
Well, before this PR, although it was `job_id`, but when this method is invoked, the input is `block_id` actually---it is just a terminology fix for this line.

In terms of whether it can be `None`, for HTEX case, it would never be. But this is a based method in `StatusHandlingExecutor`, so this is just to ensure the method is invoked in the right way here? If you think it is okay to remove this, I can remove that `if` check

Reply:
You can check how `_fail_job_async` is invoked in master at https://github.com/Parsl/parsl/blob/4f47fb4cf6ec8c7135056c4ec3a335ee6a798635/parsl/executors/high_throughput/executor.py#L625-L626